q/verbose/verbose.q

```diff
@@ -0,0 +1,471 @@
+    flip coln!colv};
+
+.finos.verbose.sym:{[str]
+    if[not 10h=type str; '".finos.verbose.sym only works on string argument"];
+    `$str};
+
+//wrapper to allow xasc in secure gateway (raw xasc may modify table in-place)
+.finos.verbose.xasc:{[sortCols;tbl]
+    if[not type[sortCols] in -11 11h; '"sort columns must be symbol(list)"];
+    if[not .Q.qt[tbl]; '".finos.verbose.xasc expects a table"];
+    sortCols xasc tbl};
+
+//wrapper to allow xdesc in secure gateway (raw xdesc may modify table in-place)
+.finos.verbose.xdesc:{[sortCols;tbl]
+    if[not type[sortCols] in -11 11h; '"sort columns must be symbol(list)"];
+    if[not .Q.qt[tbl]; '".finos.verbose.xdesc expects a table"];
+    sortCols xdesc tbl};
+
+//wrapper to allow xkey in secure gateway (raw xkey may modify table in-place)
+.finos.verbose.xkey:{[keyCols;tbl]
+    if[(not () ~ keyCols) and not type[keyCols] in -11 11h; '"sort columns must be symbol(list)"];
+    if[not .Q.qt[tbl]; '".finos.verbose.xkey expects a table"];
+    keyCols xkey tbl};
+
+//wrapper to allow case-insenstive xasc in secure gateway (raw xasc may modify table in-place)
+.finos.verbose.iasc:{[sortCols;tbl] 
+    if[not type[sortCols] in -11 11h; '"sort columns must be symbol(list)"];
+    if[not .Q.qt[tbl]; '".finos.verbose.iasc expects a table"];
+    xk:keys tbl;
+    s:((),sortCols)!{$[x in "sC";(lower;y);y]}'[(0!meta[?[tbl;();0b;{![x;x]}(),sortCols]])[;`t];sortCols];
+    .finos.verbose.xkey[xk] (0!tbl) iasc ?[tbl;();0b;s]};
+
+//wrapper to allow case-insenstive xdesc in secure gateway (raw xdesc may modify table in-place)
+.finos.verbose.idesc:{[sortCols;tbl] 
+    if[not type[sortCols] in -11 11h; '"sort columns must be symbol(list)"];
+    if[not .Q.qt[tbl]; '".finos.verbose.idesc expects a table"];
+    xk:keys tbl;
+    s:((),sortCols)!{$[x in "sC";(lower;y);y]}'[(0!meta[?[tbl;();0b;{![x;x]}(),sortCols]])[;`t];sortCols];
+    .finos.verbose.xkey[xk] (0!tbl) idesc ?[tbl;();0b;s]};
+
+//wrapper to allow null-ignoring xasc in secure gateway (raw xasc may modify table in-place)
+.finos.verbose.nasc:{[sortCols;tbl]
+    if[not type[sortCols] in -11 11h; '"sort columns must be symbol(list)"];
+    if[not .Q.qt[tbl]; '".finos.verbose.nasc expects a table"];
+    xk:keys tbl;
+    s:((),sortCols)!{$[x in "cC";y;x="s";({$[null x;::;x]}';y);(^;x$0w;y)]}'[(0!meta[?[tbl;();0b;{![x;x]}(),sortCols]])[;`t];sortCols];
+    .finos.verbose.xkey[xk] (0!tbl) iasc ?[tbl;();0b;s]
+    };
+
+//wrapper to allow null-ignoring xdesc in secure gateway (raw xasc may modify table in-place)
+.finos.verbose.ndesc:{[sortCols;tbl]
+    if[not type[sortCols] in -11 11h; '"sort columns must be symbol(list)"];
+    if[not .Q.qt[tbl]; '".finos.verbose.ndesc expects a table"];
+    sortCols xdesc tbl};
+
+// unkeys a table. It is just a projection of .finos.verbose.xkey setting the first parameter to empty list
+.finos.verbose.unkey: .finos.verbose.xkey[()];
+
+.finos.verbose.priv.validateSelectArgs:{[tbl;constr;grp;stat;cnt;ord]
```

Comment:
i would prefer the params of all the qsql wrappers conform to the traditional t/c/b/a naming; perhaps `tbl` `contr` `by` `aggs`?

Reply:
These are inherited names, but to me `stat` makes more sense because the `a` is not necessarily an aggregation.